src/test/scala/is/hail/methods/LinearRegressionSuite.scala

```diff
@@ -152,7 +152,95 @@ class LinearRegressionSuite extends SparkSuite {
+    assertDouble(qPval, v1, 0.84327106)
+
+    /*
+    v2 has two missing genotypes, comparing to output of R code as above with imputed genotypes:
+    x = c(0.9950495050004902, 1.980198019704931, 0.9950495050004902, 1.980198019704931, 0.009900990296049406, 0.009900990296049406)
+    */
+
+    assertDouble(qBeta, v2, -0.5499320)
+    assertDouble(qSe, v2, 0.3401110)
+    assertDouble(qTstat, v2, -1.616919)
+    assertDouble(qPval, v2, 0.24728705)
+
+    /*
+    v3 has two missing genotypes, comparing to output of R code as above with imputed genotypes:
+    x = c(0.009900990296049406, 0.7450495050747477, 0.9900990100009803, 0.9900990100009803, 0.9900990100009803, 0.7450495050747477)
+    */
+
+    assertDouble(qBeta, v3, 1.09536219)
+    assertDouble(qSe, v3, 0.6901002)
+    assertDouble(qTstat, v3, 1.5872510)
+    assertDouble(qPval, v3, 0.2533675)
+
+    assertEmpty(qBeta, v6)
+  }
+
+  @Test def testWithTwoCovDosage() {
+    // .gen and .sample files created from regressionLinear.vcf
+    // dosages are derived from PLs so results should agree with testWithTwoCovPhred
+    val vds = hc.importGen("src/test/resources/regressionLinear.gen", "src/test/resources/regressionLinear.sample")
+      .filterMulti()
+      .annotateSamplesTable("src/test/resources/regressionLinear.cov",
+        "Sample",
+        root = Some("sa.cov"),
+        config = TextTableConfiguration(types = Map("Cov1" -> TDouble, "Cov2" -> TDouble)))
+      .annotateSamplesTable("src/test/resources/regressionLinear.pheno",
+        "Sample",
+        root = Some("sa.pheno"),
+        config = TextTableConfiguration(types = Map("Pheno" -> TDouble), missing = "0"))
+      .linreg("sa.pheno.Pheno", Array("sa.cov.Cov1", "sa.cov.Cov2 + 1 - 1"), "va.linreg", useDosages = true, 1, 0.0)
+
+    val v1 = Variant("1", 1, "C", "T")   // x = (0, 1, 0, 0, 0, 1)
+    val v2 = Variant("1", 2, "C", "T")   // x = (., 2, ., 2, 0, 0)
+    val v3 = Variant("1", 3, "C", "T")   // x = (0, ., 1, 1, 1, .)
+    val v6 = Variant("1", 6, "C", "T")   // x = (0, 0, 0, 0, 0, 0)
+    val v7 = Variant("1", 7, "C", "T")   // x = (1, 1, 1, 1, 1, 1)
+    val v8 = Variant("1", 8, "C", "T")   // x = (2, 2, 2, 2, 2, 2)
+    val v9 = Variant("1", 9, "C", "T")   // x = (., 1, 1, 1, 1, 1)
+    val v10 = Variant("1", 10, "C", "T") // x = (., 2, 2, 2, 2, 2)
+
+    val qBeta = vds.queryVA("va.linreg.beta")._2
+    val qSe = vds.queryVA("va.linreg.se")._2
+    val qTstat = vds.queryVA("va.linreg.tstat")._2
+    val qPval = vds.queryVA("va.linreg.pval")._2
+
+    val annotationMap = vds.variantsAndAnnotations
+      .collect()
+      .toMap
 
+    def assertInt(q: Querier, v: Variant, value: Int) =
+      assert(D_==(q(annotationMap(v)).asInstanceOf[Int], value))
+
+    def assertDouble(q: Querier, v: Variant, value: Double) =
+      assert(D_==(q(annotationMap(v)).asInstanceOf[Double], value, tolerance = 1e-4))
+
+    def assertEmpty(q: Querier, v: Variant) =
+      assert(q(annotationMap(v)) == null)
```

Comment:
Don't duplicate these, pull them onto `LinearRegressionSuite` until they become useful to another test suite.